src/components/timeline/TimelineHistogram.svelte

```diff
@@ -0,0 +1,419 @@
+        }
+
+        brushed(event);
+      })
+
+      .on('end', (event: D3BrushEvent<number[]>) => {
+        brushing = false;
+        if (!event.sourceEvent) {
+          return;
+        }
+
+        let start;
+        let end;
+        if (!event.selection) {
+          const unixEpochTime = xScaleMax.invert(event.sourceEvent.offsetX).getTime();
+          const startDate = new Date(viewTimeRange.start).getTime();
+          const endDate = new Date(viewTimeRange.end).getTime();
+          const unixEpochTimeDuration = endDate - startDate;
+          // Center slider on user's mouse position
+          start = unixEpochTime - unixEpochTimeDuration / 2;
+          end = unixEpochTime + unixEpochTimeDuration / 2;
+
+          // Ensure slider is within bounds
+          const windowStartTime = xScaleMax.invert(windowMax).getTime();
+          const windowEndTime = xScaleMax.invert(windowMin).getTime();
+          if (unixEpochTimeDuration > windowEndTime - windowStartTime) {
+            // Cover case where duration could be unexpectedly large
+            start = windowStartTime;
+            end = windowEndTime;
+          } else if (start < windowStartTime) {
+            // Case where slider comes before entire window time range
+            start = windowStartTime;
+            end = start + unixEpochTimeDuration;
+          } else if (end > windowEndTime) {
+            // Case where slider comes after entire window time range
+            end = windowEndTime;
+            start = end - unixEpochTimeDuration;
+          }
+        } else {
+          start = xScaleMax.invert(event.selection[0] as number).getTime();
+          end = xScaleMax.invert(event.selection[1] as number).getTime();
+          brushed(event);
+        }
+
+        const newViewTimeRange = { end, start };
+        dispatch('viewTimeRangeChanged', newViewTimeRange);
+      });
+
+    brush = select(gTimeSelectorContainer).call(xBrush);
+    const extent = [new Date(viewTimeRange.start), new Date(viewTimeRange.end)].map(xScaleMax);
+    brush.call(xBrush.move, extent);
+  }
+
+  $: histogramHeight = (drawHeight / 5) * 2;
+  $: selectorHandleHeight = drawHeight / 1.9;
+  $: numBins = Math.max(Math.min(numBinsMax, parseInt((drawWidth / 5).toString())), numBinsMin);
+
+  // Derive activities to render in activity histogram by
+  // ORing the regexp filters of all found activity rows
+  $: if (rows) {
+    filteredActivityPoints = [];
+    // Collect all activity layers
+    const activityLayers: Layer[] = [];
+    rows.forEach(row => {
+      row.layers.forEach(layer => {
+        if (layer.chartType === 'activity') {
+          activityLayers.push(layer);
+        }
+      });
+    });
+
+    // Create an aggregate RegExp using all the filters
+    let filters = [];
+    activityLayers.forEach(l => {
+      if (l.filter && l.filter.activity?.type) {
+        filters.push(l.filter.activity?.type);
+      } else {
+        // TODO chat about what to do here
```

Comment:
@camargo Good to remove these TODOs? Wanted to make sure you took a quick look at this aspect of the activity point aggregation since I'm less familiar with all of the filtering.

Reply:
So this will actually change a bit once we change how activities are represented on the front-end. 
Is the `else` actually needed?

Reply:
TODO just remove this else